target/linux/ath79/base-files/etc/board.d/02_network

```diff
@@ -231,6 +231,10 @@ ath79_setup_interfaces()
 	tplink,tl-wr941-v2)
 		ucidef_set_interfaces_lan_wan "lan1 lan2 lan3 lan4" "wan"
 		;;
+	trendnet,tew-823dru)
+	        ucidef_add_switch "switch0" \
+			  "0@eth1" "1:lan:1" "2:lan:2" "3:lan:3" "4:lan:4" "5:wan" "6@eth0"
```

Comment:
You do not have to specify port numbers twice.
This can be merged with the definition for buffalo,bhr-4grv2